agent/agentscripts/agentscripts.go

```diff
@@ -314,6 +330,36 @@ func (r *Runner) run(ctx context.Context, script codersdk.WorkspaceAgentScript)
 		} else {
 			logger.Info(ctx, fmt.Sprintf("%s script completed", logPath), slog.F("execution_time", execTime), slog.F("exit_code", exitCode))
 		}
+
+		if r.scriptCompleted == nil {
+			logger.Debug(ctx, "r.scriptCompleted unexpectedly nil")
+			return
+		}
+
+		var stage proto.Timing_Stage
+		switch option {
+		case ExecuteStartScripts:
+			stage = proto.Timing_START
+		case ExecuteStopScripts:
+			stage = proto.Timing_STOP
+		case ExecuteCronScripts:
+			stage = proto.Timing_CRON
+		}
+
+		_, err = r.scriptCompleted(ctx, &proto.WorkspaceAgentScriptCompletedRequest{
+			Timing: &proto.Timing{
+				ScriptId:    script.ID[:],
+				DisplayName: script.DisplayName,
+				Start:       timestamppb.New(start),
+				End:         timestamppb.New(end),
+				ExitCode:    int32(exitCode),
+				Stage:       stage,
+				TimedOut:    errors.Is(err, ErrTimeout),
```

Comment:
It'd be nice to communicate ` ErrOutputPipesOpen` as well. It's a warning but not explicitly an error. Exit code could be zero but there could still be an issue like the command you had started via `mycmd &` got killed.

Say we had a status field which was `ok (!timeout && code == 0)` and `error (timeout || code != 0)`, this would be kind of like `unknown (pipes && code == 0)`.

Reply:
So would you suggest ripping out the `timed_out` column and replacing it with a `status` column that is an enum with fields:
- `ok` when `!timeout && !pipes_left_open && code == 0`
- `exit_failure` when `!timeout && !pipes_left_open && code != 0`
- `timed_out` when `timeout`
- `pipes_left_open` when `pipes_left_open`

Reply:
Will get to adding this then!

Reply:
Have made this change! Updated the description of the PR to show the new column